com.unity.shadergraph/Editor/Drawing/Views/PropertyNodeView.cs

```diff
@@ -97,6 +97,100 @@ void ChangeExposedField(bool newValue)
             UpdateIcon();
         }
 
+        internal static void AddMainColorMenuOptions(ContextualMenuPopulateEvent evt, ColorShaderProperty colorProp, GraphData graphData, Action inspectorUpdateAction)
+        {
+            if (!graphData.isSubGraph)
+            {
+                if (!colorProp.isMainColor)
+                {
+                    evt.menu.AppendAction(
+                        "Set as Main Color",
+                        e =>
+                        {
+                            foreach (var input in graphData.properties)
```

Comment:
nit: is this searching for main color and main texture something that might be worth putting GraphData?

Reply:
Probably will be worthwhile, but at the moment nothing but context menus use the maincolor/maintexture tags. Since I will be redoing the graph manipulation API, wanted to hold off on more utility functions until I had a better chance to really sit down with GraphData tho.

Reply:
Ok, this just seems like something that can easily be forgotten and duplicated over time, but I think there's no rush to do it now.